sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala

```diff
@@ -69,7 +69,7 @@ private[hive] class SparkSQLSessionManager(hiveServer: HiveServer2, sqlContext:
     setConfMap(ctx, hiveSessionState.getOverriddenConfigurations)
     setConfMap(ctx, hiveSessionState.getHiveVariables)
     if (sessionConf != null && sessionConf.containsKey("use:database")) {
-      ctx.sql(s"use ${sessionConf.get("use:database")}")
+      ctx.sessionState.catalog.setCurrentDatabase(sessionConf.get("use:database"))
```

Comment:
just to be clear, is this just a small refactoring?

Reply:
> just to be clear, is this just a small refactoring?

In this way we can avoid a lot of process part, but seems I have miss a important problem that user may write a db name wrong such as `aa.cc`

Reply:
Can we add extra tests to verify the behaviour in case of failures? Is it the same error thrown? Are we skipping some checks along the way by calling the catalog functions?